servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/AlpnClientAndServerTest.java

```diff
@@ -110,19 +112,16 @@ public AlpnClientAndServerTest(List<String> serverSideProtocols,
         });
     }
 
-    private static ServerContext startServer(List<String> supportedProtocols,
-                                             @Nullable HttpProtocolVersion expectedProtocol) throws Exception {
+    private ServerContext startServer(List<String> supportedProtocols,
+                                      @Nullable HttpProtocolVersion expectedProtocol) throws Exception {
         return HttpServers.forAddress(localAddress(0))
                 .protocols(toProtocolConfigs(supportedProtocols))
                 .secure()
                 .provider(OPENSSL)
                 .commit(DefaultTestCerts::loadServerPem, DefaultTestCerts::loadServerKey)
                 .listenBlocking((ctx, request, responseFactory) -> {
-                    if ("PRI".equals(request.method().name()) && expectedProtocol == null) {
-                        return responseFactory.badRequest();
-                    }
-                    assertThat(request.version(), is(expectedProtocol));
-                    assertThat(ctx.sslSession(), is(notNullValue()));
+                    serviceContext.set(ctx);
```

Comment:
Overwrites like this causes debugging pain if inadvertently multiple requests are sent to the service. 
I find it useful to instead use a `BlockingQueue` to track requests which gives an additional benefit of being able to wait for a request to be received by the server. However, YMMV

Reply:
Thanks for reminder, `BlockingQueue` is better!